pkgs/build-support/php/build-php.nix

```diff
@@ -0,0 +1,94 @@
+{ stdenvNoCC, lib, writeTextDir, composer, php }@inputs:
+
+{ pname
+, version
+, src
+, composerLock ? "__unset"
+, vendorDir ? "vendor"
+, vendorOutputHash
+, php ? inputs.php
+, composer ? inputs.composer
+, ...
+}@args:
+
+let
+  vendor = stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit src version;
+
+    pname = "${pname}-vendor";
+
+    # See https://github.com/NixOS/nix/issues/6660
+    dontPatchShebangs = true;
+
+    nativeBuildInputs = [
+      composer
+    ];
+
+    strictDeps = true;
+
+    buildPhase = let
+      # See https://getcomposer.org/doc/03-cli.md#composer-home-config-json
+      composerHome = let
+        content = {
+          config = {
+            autoloader-suffix = "${builtins.replaceStrings ["-"] [""] pname}";
+          };
+        };
+      in
+        writeTextDir "/config.json" (builtins.toJSON content);
+
+      composerLockFile = if true == builtins.pathExists "${src}/composer.lock"
```

Comment:
This requires building `src` before the expression can evaluate (also called [IFD](https://fzakaria.com/2020/10/21/nix-parallelism-import-from-derivation.html)). That is not allowed on Hydra.

Reply:
Ok, I'll try to find a workaround...

Reply:
Fixed.